src/components/BulkOverrideDialog/BulkOverrideInfo.js

```diff
@@ -250,6 +250,7 @@ class BulkOverrideInfo extends React.Component {
             <FormattedMessage id="ui-users.additionalInfo.placeholder">
               {placeholder => (
                 <TextArea
+                  id="data-test-additional-info"
```

Comment:
@Mathansivavelmurugesan, this is better but still not quite right. 

`data-test...` attributes are [stripped from our production build](https://github.com/folio-org/stripes-webpack/blob/04727ecb1373558ae373d41e03ef7667dfa97673/webpack.config.cli.prod.js#L114-L121). Thus, they act very much like `id` attributes in that they're unique (at least, they should be) and you can use them to make test assertions like `document.querySelector('[data-test-additional-info]')`. 

`id` attributes are useful when you need to leverage the identifier in business or style logic in the production application, e.g. to attach a new element to a specific container or set the style of a specific element, rather than just in a test scenario. Of course, you can also assert on an `id` attribute in tests, but the point is that an `id` should only be used when there is a production need for it, not just a test need. 

Here, the two have been conflated a bit: we have an `id` attribute whose value suggests it's only there for test purposes. Truly, this is no big deal. Nothing bad is gonna happen as a result. Still, it would be nice to clean up, probably replacing the `id` with `data-test-additional-info` and the test assertion with `'[data-test-additional-info]'`. 

Finally, note that the selector `#some-value` is short for `[id="some-value"]` just like `.some-class` is short for `[class="some-class"]`. They're interchangeable, but `#` and `.` are conventional. If you want pattern matching you do need to use the bracket syntax, e.g. `[class^="some-class---"]` for a class that starts with `some-class---`. Mozilla has [extensive selector documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors).